### PR TITLE
Add markdown link checker

### LINKED ISSUE #2454: Add markdown link check action(trigger on push)
We can use something like [github-action-markdown-link-check](https://github.com/gaurav-nelson/github-action-markdown-link-check) inside `create-release` or `pr-validation` workflow to confirm changes don't affect the links we have in our docs.

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2454 

**What this PR does / why we need it**:

Currently, we don't have a tool to verify the links in docs. With this PR Adding tool which can verify the links for us and make sure out docs site is always pointing to the right links. 

**Special notes for your reviewer**:
Initially, the CI would fail as will need to fix a lot of broken links. Will be checking the behaviour how it fails.

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -82,14 +82,24 @@ tasks:
...
     # disabled pending: https://github.com/wjdp/htmltest/issues/45
     # , or us moving to a no-directory URL
     env:
       NODE_PATH:
         sh: npm root -g
 
+  check-docs-site-links:
+    dir: docs/hugo
+    cmds:
+      - npm install -g markdown-link-check # TODO: Add to devcontainer?
```
Comment (theunrepentantgeek) [inline Taskfile.yml:96]: Yes, add this to the devcontainer - but leave it for a followup PR.

File: docs/hugo/link-checker.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    {
+      "pattern": "^(?!http).*"
+    }
+  ],
+  "timeout": "10s",
+  "retryOn429": true,
+  "retryCount": 3,
+  "fallbackRetryDelay": "15s",
+  "aliveStatusCodes": [200, 206, 401]
+}
```
Comment (theunrepentantgeek) [inline docs/hugo/link-checker.json:12]: Fix this.

**Review by theunrepentantgeek — APPROVED** [review]
A couple of minor fixes needed, but nothing serious. #approved

File: docs/hugo/link-checker.json (hunk 2 above)
Comment (super-harsh) [reply docs/hugo/link-checker.json]: done!

**Review by matthchr — APPROVED** [review]

**Comment by matthchr** [comment]
Very cool!

**Pull request merged**